### PR TITLE
ISPN-16278 [RESP] Fix msetnx behavior with same key

### PR DESCRIPTION
**Pull request by rigazilla**
fixes https://issues.redhat.com/browse/ISPN-16278

change map of bytes to map of wrappedbytearray so equals works and prevent entries with same key

### CODE REVIEW

File: server/resp/src/main/java/org/infinispan/server/resp/commands/string/MSETNX.java [hunk 1]
```diff
@@ -39,12 +42,14 @@ public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
          return handler.myStage();
       }
       Log.SERVER.msetnxConsistencyMessage();
-      var entries = new HashMap<byte[], byte[]>();
+      // Using WBA so equals() ensures same key has only one entry with the last value in the map
+      var entriesWBA = new HashMap<WrappedByteArray, WrappedByteArray>();
       for (int i = 0; i < arguments.size(); i++) {
-         entries.put(arguments.get(i), arguments.get(++i));
+         entriesWBA.put(new WrappedByteArray(arguments.get(i)), new WrappedByteArray(arguments.get(++i)));
```
Comment (jabolina) [inline server/resp/src/main/java/org/infinispan/server/resp/commands/string/MSETNX.java:48]: Only the keys would need to be wrapped?

**Review by jabolina — COMMENTED** [review]
Just a small change.

File: server/resp/src/main/java/org/infinispan/server/resp/commands/string/MSETNX.java (hunk 1 above)
Comment (rigazilla) [reply server/resp/src/main/java/org/infinispan/server/resp/commands/string/MSETNX.java:48]: oh yeah true, we can skip wrapping the value

**Review by jabolina — APPROVED** [review]

**Comment by jabolina** [comment]
Thanks, @rigazilla!

**Pull request merged**